Automattic/wp-calypso
# Post Editor: ditch the iframe?

This will need to be updated after #75183 and whatever follows it to enable the Site Editor un-iframing in all envs.

----
This will redirect the Editors to the URL directly rather than trying to load it in the iframe.

Enabled in dev, horizon, wpcalypso, and calypso.live environments.

Related to #74413

## Proposed Changes

* This will redirect the `/page/SITE/PAGE_ID`, `/post/SITE/POST_ID`, and `/site-editor/SITE` URLs directly to the site in question rather than iframing them. 

## Testing Instructions

* Smoke test it!
* Especially check about using an incognito window to use the various redirects to an Atomic site. Are you logged-in to the Atomic site automatically in the redirect?

## Ramifications

There are many further steps to complete this migration. The first and biggest one is testing, as **I bet that this breaks a whole bunch of e2e tests** once we turn it on for everyone.

Other things:

1. We'll want to switch to classic URLs directly in the admin menu sidebar itself. Remove for [Site Editor](https://github.com/Automattic/jetpack/blob/512b38b57af5d416043d5cfa02677f9a84a75db5/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php#L430-L448) and posts and pages are in that class too.
2. We can break these rewrites down to something that runs purely in the `/site-editor`, `post`, and `page` routes. Are there others to consider?
3. We need to fix the back button integration from the Post Editor - it's currently taking us to `wp-admin/edit.php`

## Pre-merge Checklist
